internal/kms/pubsub.go

```diff
@@ -260,9 +279,12 @@ func (s *pubSubService) tryGenEncryptionKeyLocally(
 	req *fetchEncryptionKeyRequest,
 ) (*fetchEncryptionKeyReply, error) {
 	blocks, err := s.getEncryptionKeysLocally(ctx, req)
-	if err != nil || len(blocks) == 0 {
+	if err != nil {
 		return nil, err
 	}
+	if len(blocks) == 0 {
+		return &fetchEncryptionKeyReply{}, nil
+	}
```

Comment:
suggestion: Maybe hit with a test if possible.

Reply:
I added an additional tests. But strangely that line was already hit by another test

Reply:
If it was unexpectedly hit it would be worth finding out why.

Reply:
it is expected: upon doc creation the peer that receive an update (not creator) requests encryption key as it can't merge blocks. But the creator peer rejects (by responding with empty response), because the access was not granted yet.

Reply:
Thanks for the explanation Islam :)